packages/neon_framework/lib/src/widgets/dialog.dart

```diff
@@ -523,3 +555,395 @@ class NeonUnifiedPushDialog extends StatelessWidget {
+          ],
+        );
+      },
+    );
+  }
+
+  Widget _buildPredefinedStatuses() => ResultBuilder.behaviorSubject(
+        subject: bloc.predefinedStatuses,
+        builder: (context, predefinedStatusesResult) => ResultBuilder.behaviorSubject(
+          subject: bloc.status,
+          builder: (context, statusResult) {
+            final selectedPredefinedStatus = statusResult.hasData && statusResult.requireData is user_status.Private
+                ? (statusResult.requireData as user_status.Private).messageId
+                : null;
+
+            final tiles = predefinedStatusesResult.data
+                ?.map(
+                  (status) => Card(
+                    shape: selectedPredefinedStatus == status.id ? selectedBorder : null,
+                    child: InkWell(
+                      borderRadius: const BorderRadius.all(Radius.circular(12)),
+                      onTap: () {
+                        bloc.setPredefinedMessage(
+                          id: status.id,
+                          clearAt: status.clearAt != null
+                              ? status.clearAt.toDateTime()!.millisecondsSinceEpoch ~/ 1000
+                              : null,
+                        );
+                      },
+                      child: ListTile(
+                        leading: Text(
+                          status.icon,
+                          style: const TextStyle(fontSize: 18),
+                        ),
+                        title: Text(status.message),
+                        subtitle: Text(status.clearAt.formatRelative(context)),
+                      ),
+                    ),
+                  ),
+                )
+                .toList();
+
+            return Column(
+              children: [
+                NeonError(
+                  predefinedStatusesResult.error,
+                  onRetry: bloc.refresh,
+                ),
+                NeonLinearProgressIndicator(
+                  visible: predefinedStatusesResult.isLoading,
+                ),
+                if (tiles != null) buildGrid(NeonDialogTheme.of(context).constraints, tiles),
+              ],
+            );
+          },
+        ),
+      );
+
+  ShapeBorder get selectedBorder => RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(12),
+        side: BorderSide(
+          color: Theme.of(context).colorScheme.primary,
+          width: 1.5,
+        ),
+      );
+
+  Widget buildGrid(
+    BoxConstraints constraints,
+    List<Widget> children,
+  ) =>
+      SizedBox(
+        height: 72 * 3, // TODO: This needs to go away to allow dynamic sizing.
+        child: Wrap(
+          children: children
+              .map(
+                (child) => SizedBox(
+                  height: 72,
+                  width: constraints.maxWidth / 2,
+                  child: child,
+                ),
+              )
+              .toList(),
+        ),
+      );
```

Comment:
Should both be handled by the common widget.

Reply:
wdym?

Reply:
https://github.com/nextcloud/neon/pull/1378#discussion_r1439063763